Samples/AudioPreprocessing/AudioPreprocessing/Model/PreprocessModel.cs

```diff
@@ -1,16 +1,323 @@
+                .SetInput("data", "power_frames")
+                .SetConstant("shape", TensorInt64Bit.CreateFromArray(new List<long>() { 2 }, new long[] { batchSize * nDfts, onesidedDftSize }))
+                .SetOutput("reshaped", "reshaped_output"))
+              .Operators.Add(new Operator("MatMul")
+                .SetInput("A", "reshaped_output")
+                .SetInput("B", "mel_weight_matrix")
+                .SetOutput("Y", "mel_spectrogram"))
+              .Operators.Add(new Operator("Reshape")
+                .SetInput("data", "mel_spectrogram")
+                .SetConstant("shape", TensorInt64Bit.CreateFromArray(new List<long>() { 4 }, melSpectrogramShape))
+                .SetOutput("reshaped", "Output.MelSpectrogram"));
+
+            var model = builder.CreateModel();
+
+            LearningModelSession session = new LearningModelSession(model);
+            LearningModelBinding binding = new LearningModelBinding(session);
+
+            binding.Bind("Input.TimeSignal", TensorFloat.CreateFromArray(signalShape, signal.ToArray()));
+
+            // Bind output
+            var outputImage = new VideoFrame(
+                BitmapPixelFormat.Bgra8,
+                nMelBins,
+                nDfts);
+            //var outputImage = new Bitmap(
+            //    nMelBins,
+            //    nDfts);
+
+
+            binding.Bind("Output.MelSpectrogram", outputImage);
+
+            // Evaluate
+            var sw = Stopwatch.StartNew();
+            var result = session.Evaluate(binding, "");
+            sw.Stop();
+            Console.WriteLine("Evaluate Took: %f\n", sw.ElapsedMilliseconds);
+
+            //return outputImage.SoftwareBitmap;
+            return outputImage.SoftwareBitmap;
+
+            // check the output video frame object by saving output image to disk
+            //string imagename = "mel_spectrogram.jpg";
+
+            //// save the output
+
+            //string modulepath = filehelpers.getmodulepath();
+            //winrt.windows.storage.storagefolder folder = winrt.windows.storage.storagefolder.getfolderfrompathasync(modulepath).get();
+            //winrt.windows.storage.storagefile file = folder.createfileasync(out_name, winrt.windows.storage.creationcollisionoption.replaceexisting).get();
+            //winrt.windows.storage.streams.irandomaccessstream write_stream = file.openasync(winrt.windows.storage.fileaccessmode.readwrite).get();
+            //winrt.windows.graphics.imaging.bitmapencoder encoder = winrt.windows.graphics.imaging.bitmapencoder.createasync(winrt.windows.graphics.imaging.bitmapencoder.jpegencoderid(), write_stream).get();
+            //encoder.setsoftwarebitmap(outputimage.softwarebitmap);
+            //encoder.flushasync().get();
+
+            // save the model
+            //builder.save("spectrogram.onnx");
+            //return "todo: replace with image path? done melspectrogram generation";
+        }
+
+        static void ModelBuilding_MelWeightMatrix()
```

Comment:
You dont need these other static methods right?

Reply:
Removed